tests/validation/tests/v3_api/test_node_label.py

```diff
@@ -0,0 +1,1178 @@
+    client.update(node, labels=node_labels)
+
+
+def test_node_label_custom_add_edit_addnode():
+    """ Create a custom cluster, Add node labels via register command
+    Edit via API and change the existing label value ONLY
+    Add a control plane node with label same as the ORIGINAL one
+    And check the labels on all the nodes. This will fail due to issue:
+    https://github.com/rancher/rancher/issues/25144"""
+    test_label = random_name()
+    label_value = random_name()
+    cluster_custom["test_label"] = test_label
+    cluster_custom["label_value"] = label_value
+    client = cluster_detail["client"]
+    node_roles = [["worker", "controlplane", "etcd"]]
+    aws_nodes_list = []
+    cluster, aws_nodes = \
+        create_custom_node_label(node_roles, test_label, label_value, True)
+    create_kubeconfig(cluster)
+    for node in aws_nodes:
+        aws_nodes_list.append(node)
+
+    nodes = client.list_node(clusterId=cluster.id).data
+    node = nodes[0]
+
+    validate_label_set_on_node(client, node, test_label, label_value)
+
+    node_name_1 = node.nodeName
+    # edit label through API
+    node = client.reload(node)
+    node_labels = node.labels.data_dict()
+    new_value_2 = random_name()
+    node_labels[test_label] = new_value_2
+    client.update(node, labels=node_labels)
+    node = client.reload(node)
+    time.sleep(2)
+
+    # Label should be added
+    wait_for_condition(client, node, check_label_added(test_label), None, 10)
+    validate_label_set_on_node(client, node, test_label, new_value_2)
+
+    # add a control plane node with original label
+    aws_nodes = \
+        aws_nodes = \
+        AmazonWebServices().create_multiple_nodes(
+            1, random_test_name(HOST_NAME))
+    for node in aws_nodes:
+        aws_nodes_list.append(node)
+
+    aws_node = aws_nodes[0]
+    docker_run_cmd = get_custom_host_registration_cmd(client, cluster,
+                                                      ["controlplane"],
+                                                      aws_node)
+    docker_run_cmd = docker_run_cmd + " --label " + \
+                    test_label + "=" + label_value
+    aws_node.execute_command(docker_run_cmd)
+    wait_for_cluster_node_count(client, cluster, 2)
+    cluster = validate_cluster_state(client, cluster,
+                                     intermediate_state="updating")
+    nodes = client.list_node(clusterId=cluster.id).data
+    # cluster cleanup
+    custom_cluster_add_edit["cluster"] = cluster
+    custom_cluster_add_edit["aws_node"] = aws_nodes_list
+    
+    for node in nodes:
+        if node.nodeName != node_name_1:
+            validate_label_set_on_node(client, node, test_label, label_value)
+        else:
+            validate_label_set_on_node(client, node, test_label, new_value_2)
+
+
+def test_node_label_node_template_add():
+    """
+    This test validates label added through node template,
+    delete label on node template, and validates the label
```

Comment:
It looks like this test creates a new node template with a new label, not delete label on the existing node template, right? Can you update the description?

Reply:
Done